src/ledger/fuzz.zig

```diff
@@ -0,0 +1,330 @@
+
+fn dbDeleteFilesInRange(
+    db: *BlockstoreDB,
+    random: *const std.rand.Random,
+    count: *std.atomic.Value(u64),
+    max_actions: ?usize,
+) !void {
+    const start = random.int(u32);
+    const end = blk: {
+        const end_ = random.int(u32);
+        if (end_ < start)
+            break :blk (end_ +| start)
+        else
+            break :blk end_;
+    };
+
+    try performDbAction(
+        "RocksDb.deleteFilesInRange",
+        BlockstoreDB.deleteFilesInRange,
+        .{ db, cf1, start, end },
+        count,
+        max_actions,
+    );
+}
+
+fn dbGetBytes(
+    db: *BlockstoreDB,
+    random: *const std.rand.Random,
+    count: *std.atomic.Value(u64),
+    max_actions: ?usize,
+) !void {
+    const key = random.int(u32);
+    try performDbAction(
+        "RocksDb.getBytes",
+        BlockstoreDB.getBytes,
+        .{ db, cf1, key },
+        count,
+        max_actions,
+    );
+}
+
+fn dbGet(
+    db: *BlockstoreDB,
+    random: *const std.rand.Random,
+    count: *std.atomic.Value(u64),
+    max_actions: ?usize,
+) !void {
+    const key = random.int(u32);
+    try performDbAction(
+        "RocksDb.get",
+        BlockstoreDB.get,
+        .{ db, allocator, cf1, key },
+        count,
+        max_actions,
+    );
+}
+
+fn dbCount(
+    db: *BlockstoreDB,
+    count: *std.atomic.Value(u64),
+    max_actions: ?usize,
+) !void {
+    try performDbAction(
+        "RocksDb.count",
+        BlockstoreDB.count,
+        .{ db, cf1 },
+        count,
+        max_actions,
+    );
+}
+
+fn dbContains(
+    db: *BlockstoreDB,
+    random: *const std.rand.Random,
+    count: *std.atomic.Value(u64),
+    max_actions: ?usize,
+) !void {
+    const key = random.int(u32);
+    try performDbAction(
+        "RocksDb.contains",
+        BlockstoreDB.contains,
+        .{ db, cf1, key },
+        count,
+        max_actions,
+    );
+}
+
+// Batch API
+fn batchDeleteRange(
+    db: *BlockstoreDB,
+    random: *const std.rand.Random,
+    count: *std.atomic.Value(u64),
+    max_actions: ?usize,
+) !void {
+    var last_print_msg_count: u64 = 0;
+    while (true) {
+        if (max_actions) |max| {
+            if (count.load(.monotonic) >= max) {
+                std.debug.print("Batch actions reached max actions: {}\n", .{max});
+                break;
+            }
+        }
+        const start = random.int(u32);
+        const end = blk: {
+            const end_ = random.int(u32);
+            if (end_ < start)
+                break :blk (end_ +| start)
+            else
+                break :blk end_;
+        };
+
+        const key = random.int(u32);
+        var buffer: [61]u8 = undefined;
+
+        // Fill the buffer with random bytes
+        for (0..buffer.len) |i| {
+            buffer[i] = @intCast(random.int(u8));
+        }
+
+        const value: []const u8 = buffer[0..];
+
+        var batch = try db.initWriteBatch();
+        defer batch.deinit();
+
+        try batch.put(cf1, key, Data{ .value = value });
```

Comment:
why have a put in a delete range?

Reply:
This is because the batch API is being performed, which involves batching the put and the delete actions.